collect_app/src/main/java/org/odk/collect/android/utilities/AppStateProvider.kt

```diff
@@ -1,25 +0,0 @@
-package org.odk.collect.android.utilities
-
-import android.content.pm.PackageInfo
-import org.odk.collect.android.preferences.keys.MetaKeys
-import org.odk.collect.shared.Settings
-
-class AppStateProvider(private val packageInfo: PackageInfo, private val metaSettings: Settings) {
-
-    fun isFreshInstall(): Boolean {
-        return !isUpdatedVersion() && !metaSettings.contains(MetaKeys.FIRST_LAUNCH)
-    }
-
-    private fun isUpdatedVersion(): Boolean {
-        return if (alwaysFresh) {
-            false
-        } else {
-            packageInfo.firstInstallTime != packageInfo.lastUpdateTime
```

Comment:
This logic wouldn't have worked in cases where we upgrade and then clear state - we'd really want to treat that as a "fresh install". Again I could probably have held back on addressing this case here, but I found it very hard to keep my mind straight on everything else knowing about these little edge cases.